infra/feast-operator/internal/controller/services/repo_config_test.go

```diff
@@ -201,3 +281,37 @@ func minimalFeatureStore() *feastdevv1alpha1.FeatureStore {
 		},
 	}
 }
+
+func emptyMockExtractConfigFromSecret(secretRef string, secretKeyName string) (map[string]interface{}, error) {
+	return map[string]interface{}{}, nil
+}
+
+func mockExtractConfigFromSecret(secretRef string, secretKeyName string) (map[string]interface{}, error) {
+	return createParameterMap(), nil
+}
+
+func createParameterMap() map[string]interface{} {
+	yamlString := `
+hosts:
+  - 192.168.1.1
+  - 192.168.1.2
+  - 192.168.1.3
+keyspace: KeyspaceName
+port: 9042                                                              
+username: user                                                          
+password: secret                                                        
+protocol_version: 5                                                     
+load_balancing:                                                         
+  local_dc: datacenter1                                             
+  load_balancing_policy: TokenAwarePolicy(DCAwareRoundRobinPolicy)
+read_concurrency: 100                                                   
+write_concurrency: 100
+`
```

Comment:
thoughts on adding an incorrect `type` here so we can unit test this scenario?
```suggestion
func createParameterMap() map[string]interface{} {
	yamlString := `
type: wrong
hosts:
  - 192.168.1.1
  - 192.168.1.2
  - 192.168.1.3
keyspace: KeyspaceName
port: 9042                                                              
username: user                                                          
password: secret                                                        
protocol_version: 5                                                     
load_balancing:                                                         
  local_dc: datacenter1                                             
  load_balancing_policy: TokenAwarePolicy(DCAwareRoundRobinPolicy)
read_concurrency: 100                                                   
write_concurrency: 100
`
```

Reply:
Tracked new GH issue with leftovers